### PR TITLE
APPIOS-2351: Added wallet statements endpoint

### PR DESCRIPTION
**Pull request by softwaresaiyajin**

### CODE REVIEW

File: Sources/Services/WalletAsyncClient/WalletAsyncClient.swift [hunk 1]
```diff
@@ -183,4 +183,20 @@ public class WalletAsyncClient: BaseAsyncClient {
     public func deleteSubscribers() -> Promise<[PSSubscriber]> {
         return doRequest(requestRouter: WalletApiRequestRouter.deleteSubscribers)
     }
+    
+    public func getWalletStatements(
```
Comment (vytautasgimbutas) [inline Sources/Services/WalletAsyncClient/WalletAsyncClient.swift:187]:
```
getStatements(filter: PSStatementFilter)

class PSStatementFilter(
  walletId: Int
  from: Int,
  to: Int
) : PSBaseFilter

File: Sources/Entities/Wallet/PSWalletStatement.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import ObjectMapper
+
+public class PSWalletStatement: Mappable {
```
Comment (vytautasgimbutas) [inline Sources/Entities/Wallet/PSWalletStatement.swift:3]: PSStatement
PSOtherParty (as can be reused with PSPendingPayment, PSTransactionRequest entities)

File: Sources/Entities/Wallet/PSWalletStatement.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public var date: Int!
+    public var amount: Int!
+    public var direction: String!
+    public var details: String?
+    public var type: String!
+    public var currency: String!
+    public var otherParty: PSWalletStatementOtherParty?
+    
+    required public init?(map: Map) { }
+
```
Comment (vytautasgimbutas) [inline Sources/Entities/Wallet/PSWalletStatement.swift:18]: convenience method
```
getAmountMoney() -> PSMoney { return *from amountDecimal and currency*}

File: Sources/Entities/PSOtherParty.swift [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import ObjectMapper
+
+public class PSOtherParty: Mappable {
+    
+    public var displayName: String!
```
Comment (vytautasgimbutas) [inline Sources/Entities/PSOtherParty.swift:5]: All of them can be optional

File: Sources/Entities/PSStatementFilter.swift [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import PayseraCommonSDK
+import ObjectMapper
+
+public class PSStatementFilter: PSBaseFilter {
+    
+    public let walletId: Int
```
Comment (vytautasgimbutas) [inline Sources/Entities/PSStatementFilter.swift:6]: All of them can be optional

File: Sources/Services/WalletAsyncClient/WalletAsyncClient.swift (hunk 1 above)
Comment (softwaresaiyajin) [reply Sources/Services/WalletAsyncClient/WalletAsyncClient.swift]: Done

File: Sources/Entities/Wallet/PSWalletStatement.swift (hunk 2 above)
Comment (softwaresaiyajin) [reply Sources/Entities/Wallet/PSWalletStatement.swift]: Done

File: Sources/Entities/Wallet/PSWalletStatement.swift (hunk 3 above)
Comment (softwaresaiyajin) [reply Sources/Entities/Wallet/PSWalletStatement.swift]: Done

File: Sources/Entities/PSStatementFilter.swift [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    public let fromDate: Int
```
Comment (vytautasgimbutas) [inline Sources/Entities/PSStatementFilter.swift:7]:
```
from: Date
to: Date
```
use DateTransformer from swift-lib-common-sdk

File: Sources/Entities/PSOtherParty.swift (hunk 4 above)
Comment (softwaresaiyajin) [reply Sources/Entities/PSOtherParty.swift]: Done

File: Sources/Entities/PSStatementFilter.swift (hunk 5 above)
Comment (softwaresaiyajin) [reply Sources/Entities/PSStatementFilter.swift]: Done

File: Sources/Entities/PSStatementFilter.swift (hunk 6 above)
Comment (softwaresaiyajin) [reply Sources/Entities/PSStatementFilter.swift]: Done

File: Sources/Entities/PSStatementFilter.swift [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+import ObjectMapper
+
+public class PSStatementFilter: PSBaseFilter {
+    
+    public var from: Date?
+    public var to: Date?
+    
+    public func toJSON() -> [String : Any] {
+        var parameters = super.toJSON()
+        if let value = DateTransform().transformToJSON(from) {
```
Comment (vytautasgimbutas) [inline Sources/Entities/PSStatementFilter.swift:11]: You can override mapping function and use in same manner as:
{code}
usedAt <- (map["used_at"], DateTransform())
{code}

**Pull request merged**